lib/indexer/filterindexer.js

```diff
@@ -59,12 +75,16 @@ class FilterIndexer extends Indexer {
   async indexBlock(meta, block, view) {
     const hash = block.hash();
     const prev = await this.getFilterHeader(block.prevBlock);
-    const basic = block.toFilter(view);
+    const gcsFilter = block.toFilter(view, this.filterType);
 
     const filter = new Filter();
-    filter.header = basic.header(prev);
-    filter.filter = basic.toRaw();
-    await this.blocks.writeFilter(hash, filter.toRaw());
+    filter.header = gcsFilter.header(prev);
+    filter.filter = gcsFilter.toRaw();
+
+    await this.blocks.writeFilter(hash, filter.toRaw(), this.filterType);
+
+    const filterHash = hash256.digest(filter.filter);
```

Comment:
We shouldn't have to hash this data (again) right here. On line 81 we call `header()`:

https://github.com/bcoin-org/bcoin/blob/ab4197d2f6faf64eac9afa633f57e28136e2da7b/lib/golomb/golomb.js#L70-L72

...which calls hash():

https://github.com/bcoin-org/bcoin/blob/ab4197d2f6faf64eac9afa633f57e28136e2da7b/lib/golomb/golomb.js#L58-L61

I think we can add a cache property `this._hash` to `class Golomb` (you can see this kind of pattern in other bcoin objects like TX and Block). Then the function hash() checks the cache before recalculating. We can then just use `gcsFilter.hash()` here.